docs/index.rst

```diff
@@ -36,20 +65,107 @@ The plugin may be used within a north *task* or *service*. Both of these are cre
+
+  - **Username**: Specify the username to use for authentication. This is only used if the *User Authentication Mode* is set to *Username And Password*.
+
+  - **Password**: Specify the password to use for authentication. This is only used if the *User Authentication Mode* is set to *Username And Password*.
+
+  - **Server Public Certificate**: (Optional) The name of the public certificate of the OPC/UA server specified in the *OPCUA Server URL*. This must either be a DER or PEM format certificate file.
+
+  - **Client Public Certificate**: The name of the public certificate of the OPC/UA client application, that is, this plugin. This must either be a DER or PEM format certificate file.
+
+  - **Client Private Key**: The name of the private key of the client application, that is, the private key the plugin will use. This must be a PEM format key file.
+
+  - **Client Private Passphrase Key**: (Optional) The passphrase of the private key of the client application.
+
+
+- Click *Next*
+
+- Enable your task or service
+
+- Click *Done*
+
+
+Certificate Management
+----------------------
+
+Typically the Certificate Authorities certificate is retrieved and uploaded to the Fledge Certificate Store along with the certificate from the OPC/UA server that has been signed by that Certificate Authority. A public/private key pair must also be created for the plugin and signed by the Certificate Authority. These are uploaded to the Fledge Certificate Store.
+
+|OpenSSL| may be used to generate and convert the keys and certificates required.
+An example to |generate_certificate| to do this is available as part of the underlying |FreeOpcUa|.
+
+.. note::
+    subjectAltName should be same as your OPCUA server Application URI.
+
+Use the OpenSSL command-line utility to generate the certificate and key files to the formats needed for the Plugin.
+
+.. code-block:: bash
+
+   openssl genrsa -out myclientkey.pem 2048
+
+   openssl req -x509 -days 365 -new -out myclientcert.pem -key myclientkey.pem -config ssl.conf
+
+   openssl x509 -outform der -in myclientcert.pem -out myclientcert.der
+
+Importing the Certificate and Key Files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can either use GUI or manually to upload the certificates at right place.
+
+a) via GUI (Recommended way)
+
+- Launch the Fledge GUI.
+- Navigate to the Certificate Store.
+- In the upper right corner of the screen, click *Import*.
+
+    +-------------+
+    | |certstore| |
+    +-------------+
+
+    - In the *Key* section, click *Choose File* and navigate to the location of the key file *myclientkey.pem*.
+
+    - In the *Certificate* section, click *Choose File* and navigate to the location of the certificate file *myclientcert.der*.
+
+    - Click *Import*.
+
+And for the OPC UA server certificate, enter the server certificate file name in the *Certificate* portion of the Import dialog and then click *Import*.
 
-  - Enable your task or service and click *Done*
+b) via Manually
```

Comment:
No need to use "via." Say "Using the Fledge GUI" for part 1, and "Manually" for part 2.